pct.sh

```diff
@@ -103,26 +103,39 @@ java \
 	-skipTestCache true
 
 if grep -q -F -e '<status>INTERNAL_ERROR</status>' pct-report.xml; then
-	echo PCT failed
+	echo 'PCT failed with internal error' >&2
+	cat pct-report.xml
+	exit 1
+elif grep -q -F -e '<status>COMPILATION_ERROR</status>' pct-report.xml; then
+	echo 'PCT failed with compilation error' >&2
 	cat pct-report.xml
 	exit 1
 elif grep -q -F -e '<status>TEST_FAILURES</status>' pct-report.xml; then
-	echo PCT marked failed, checking to see if that is due to a failure to run tests at all
+	#
+	# Previous versions of PCT claimed that there were test failures even when no tests had been
+	# run at all. While it is possible that current versions of PCT no longer exhibit this
+	# pathology, we err on the side of caution and check anyway.
+	#
+	echo 'PCT marked failed, checking to see if that is due to a failure to run tests at all' >&2
```

Comment:
I thought about deleting this logic entirely, but I failed to convince myself that it was not needed, so better safe than sorry.